lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -4,163 +4,446 @@
+        });
+
+        it('sets the accessible name from the row cell contents, separated by a semi-colon', function() {
+          const cell = renderTableCell({
+            rowBtnIcon: faEdit,
+            sortDir: 'asc'
+          });
+
+          expect(within(cell).getByRole('button')).toHaveAccessibleName('Foo; Bar');
+        });
       });
     });
-    component.update();
 
-    expect(component.find('td')).toHaveProp('colSpan', 1);
-    expect(component.find('td')).toHaveClassName('nx-cell--meta-info');
-    expect(component.find('td')).toIncludeText('Error!');
+    describe('When isSortable is true', function() {
+      it('sets default aria-sort', function() {
+        expect(renderColumnHeader({ isSortable: true })).toHaveAttribute('aria-sort', 'none');
+      });
+
+      it('sets ascending aria-sort', function() {
+        expect(renderColumnHeader({ isSortable: true, sortDir: 'asc' })).toHaveAttribute('aria-sort', 'ascending');
+      });
+
+      it('sets descending aria-sort', function() {
+        expect(
+            renderColumnHeader({ isSortable: true, sortDir: 'desc' })
+        ).toHaveAttribute('aria-sort', 'descending');
+      });
+
+      it('renders default tooltip and content', async function() {
+        const user = userEvent.setup();
+
+        const sortDefault = within(
+            renderColumnHeader({ isSortable: true, children: 'foo' })!
+        ).getByRole('button');
+
+        await user.hover(sortDefault);
+        await runTimers();
+        const tooltip = await screen.findByRole('tooltip');
+        expect(tooltip).toBeInTheDocument();
+        expect(tooltip).toHaveTextContent('foo unsorted');
+      });
+
+      it('renders ascending tooltip and content', async function() {
+        const user = userEvent.setup();
+
+        const sortAsc = within(
+            renderColumnHeader({ isSortable: true, children: 'foo', sortDir: 'asc' })!
+        ).getByRole('button');
+
+        await user.hover(sortAsc);
+        await runTimers();
+        const tooltip = await screen.findByRole('tooltip');
+        expect(tooltip).toBeInTheDocument();
+        expect(tooltip).toHaveTextContent('foo ascending');
+      });
+
+      it('renders ascending tooltip and content', async function() {
+        const user = userEvent.setup();
+
+        const sortDesc = within(
+            renderColumnHeader({ isSortable: true, children: 'foo', sortDir: 'desc' })!
+        ).getByRole('button');
+
+        await user.hover(sortDesc);
+        await runTimers();
+        const tooltip = await screen.findByRole('tooltip');
+        expect(tooltip).toBeInTheDocument();
+        expect(tooltip).toHaveTextContent('foo descending');
+      });
+
+      it('sets default button accessible name to match the tooltip', function() {
+        expect(
+            within(renderColumnHeader({ isSortable: true, children: 'foo' })
+            ).getByRole('button'))
+            .toHaveAccessibleName('foo unsorted');
```

Comment:
Weird formatting in these three tests. I'd pull the element selection (`within(...).getByRole(...)`) out into a variable.